### PR TITLE
Exceptions Handling added

### PR DESCRIPTION
**Pull request by NataliaCarvalho03**
Exception Handling added to the mosaic_creator.py and low_poly.py code files. The objective is give to the user clear information about the occurred error and ignore non-image files present in the directory passed to the mosaic_creator.py algorithm.

### CODE REVIEW

File: mosaic_creator.py [hunk 1]
```diff
@@ -11,17 +12,39 @@
...
+                #If cant open the file, ignore it and continue
+                print("WARNING: Can not identify and open", file, "Ignoring this file...")
+                continue
+            except IsADirectoryError:
+                print("Argument passed to --images is a directory without any files. Image files path expected.")
+                sys.exit()
+        
+        #When there is no image files in the directory, but it has some other files (pdf, txt, etc), 
+        #files list will be empty
+        if images:
```
Comment (smaranjitghose) [inline mosaic_creator.py]: This is not required. We can simply use glob to check the number of files with *png or *jpg in the given path

File: mosaic_creator.py [hunk 2]
```diff
@@ -11,17 +12,39 @@
...
+            return (images)
+        else:
+            print("There is no valid image file in the directory passed as argument to --images.")
+            sys.exit()
+    else:
```
Comment (smaranjitghose) [inline mosaic_creator.py]: Same as above

File: mosaic_creator.py [hunk 3]
```diff
@@ -77,11 +100,22 @@ def createPhotomosaic(target_image, input_images, grid_size,
...
+        except TypeError:
+            print("Warning: The length of 1D weights is not the same as the shape of a along axis. Ignoring", img, "file.")
             continue
 
     for img in tqdm(target_images):
-        avg = getAverageRGB(img)
+        try:
+            avg = getAverageRGB(img)
+        except ZeroDivisionError:
+            print("WARNING: Division by zero detected. Ignoring", img, "file.")
```
Comment (smaranjitghose) [inline mosaic_creator.py]: Use F Strings.
Additionally there is no need to print the image array

File: mosaic_creator.py [hunk 4]
```diff
@@ -92,10 +126,27 @@ def createPhotomosaic(target_image, input_images, grid_size,
...
+	sys.exit()
+except IsADirectoryError:
+    print("The argument passed as --target is a directory without any files. Image file path expected.")
+    sys.exit()
 
-target_image = Image.open(args.target)
 # input images
-input_images = getImages(args.images)
+try:
+    input_images = getImages(args.images)
```
Comment (smaranjitghose) [inline mosaic_creator.py]: For each line we do not need a new try except clause. This can be included in the last one as well.

File: low_poly.py [hunk 5]
```diff
@@ -33,7 +49,14 @@
 vertices.append((rows - 1, 0))
 vertices.append((rows - 1, cols - 1))
 
-triangulation = Delaunay(vertices)
+try:
+	triangulation = Delaunay(vertices)
+except QhullError:
```
Comment (smaranjitghose) [inline low_poly.py]: We can use a single try-except block to handle all the errors.

**Review by smaranjitghose — CHANGES_REQUESTED** [review]

File: mosaic_creator.py [hunk 6]
```diff
@@ -53,7 +69,7 @@ def getBestMatchIndex(input_avg, avgs):
             min_index = index
         index += 1
     return (min_index)
-# creating the image grid for tiles to fit in
+# creating the image grid for tilehttps://twitter.com/notificationss to fit in
```
Comment (smaranjitghose) [inline mosaic_creator.py]: What's this about😅

**Review by smaranjitghose — APPROVED** [review]

File: mosaic_creator.py (hunk 6 above)
Comment (NataliaCarvalho03) [reply mosaic_creator.py]: OMG! I'm sorry for this mistake, I didn't realize it. Thanks for having patience and for helping me to improve my skills.

**Pull request merged**